apple/internal/partials/debug_symbols.bzl

```diff
@@ -101,21 +105,41 @@ def _bundle_dsym_files(ctx, debug_provider, bundle_name, bundle_extension = ""):
     bundle_name_with_extension = bundle_name + bundle_extension
     dsym_bundle_name = bundle_name_with_extension + ".dSYM"
 
-    outputs = []
+    output_binary = ctx.actions.declare_file(
+        "%s/Contents/Resources/DWARF/%s" % (
+            dsym_bundle_name,
+            bundle_name,
+        ),
+    )
+    outputs = [output_binary]
+
+    inputs = []
+    args = [
+        "lipo",
+        "-output",
+        output_binary.path,
+        "-create",
+    ]
 
     # TODO(b/36174487): Iterate over .items() once the Map/dict problem is fixed.
     for arch in debug_provider.outputs_map:
         arch_outputs = debug_provider.outputs_map[arch]
         dsym_binary = arch_outputs["dsym_binary"]
-        output_binary = ctx.actions.declare_file(
-            "%s/Contents/Resources/DWARF/%s_%s" % (
-                dsym_bundle_name,
-                bundle_name,
-                arch,
-            ),
-        )
-        outputs.append(output_binary)
-        ctx.actions.symlink(target_file = dsym_binary, output = output_binary)
+        inputs.append(dsym_binary)
+        args.extend([
+            "-arch",
+            arch,
+            dsym_binary.path,
+        ])
+
+    apple_support.run(
```

Comment:
in the case that this is only built for a single arch, should we avoid this lipo and do a cp?